### PR TITLE
[docs] Add missing descriptions in `GridFilterApi`

### PR DESCRIPTION
**Pull request by m4theushw**
The methods from `GridFilterApi` in https://deploy-preview-1529--material-ui-x.netlify.app/api/data-grid/grid-api/ don't have a description.

### CODE REVIEW

File: packages/grid/_modules_/grid/models/api/gridFilterApi.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   * @param item
+   * @param linkOperator
+   */
+  applyFilter: (item: GridFilterItem, linkOperator?: GridLinkOperator) => void;
+  /**
+   * Applies all filters on all rows.
+   */
+  applyFilters: () => void;
+  /**
+   * Deletes a [[GridFilterItem]]
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/models/api/gridFilterApi.ts:34]:
```suggestion
   * Deletes a [[GridFilterItem]].
```

File: packages/grid/_modules_/grid/models/api/gridApi.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export interface GridApi
     GridPaginationApi,
     GridCsvExportApi,
     GridFocusApi,
-    FilterApi,
+    GridFilterApi,
     ColumnMenuApi,
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/models/api/gridApi.ts:41]: **Side**

We have the same wrong naming here:

```suggestion
    GridColumnMenuApi,
```

File: packages/grid/_modules_/grid/models/api/gridFilterApi.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { GridRowId, GridRowModel } from '../gridRows';
+import { GridFilterModel } from '../../hooks/features/filter/gridFilterModelState';
+import { GridFilterItem, GridLinkOperator } from '../gridFilterItem';
+
+/**
+ * The filter API interface that is available in the grid [[apiRef]].
+ */
+export interface GridFilterApi {
+  /**
+   * Shows the filter panel. If `targetColumnField` is given, a filter for this field is also added.
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/models/api/gridFilterApi.ts:10]: **Side**

Great initiative with the plural, most of the other descriptions are singular, which is wrong.

**Review by dtassone — APPROVED** [review]

**Review by DanailH — APPROVED** [review]

File: packages/grid/_modules_/grid/models/api/gridApi.ts (hunk 2 above)
Comment (oliviertassinari) [reply packages/grid/_modules_/grid/models/api/gridApi.ts:41]: I have created https://github.com/mui-org/material-ui-x/issues/1705 so we don't forget. We can come back to it at some point in the future.

**Pull request merged**